File: ArtSiteWeb/artsite-web/src/network/models/searchModels/SearchObject.ts
import { Gender } from "src/network/enums/gender";
import { Sexuality } from "src/network/enums/sexuality";

export class SearchObject{
  public nameString?: string;
  public tags?: Map<number, boolean>;
  public minAge?: number;
  public maxAge?: number;
  public species?: number;
  public gender?: Gender;
  public sexuality?: Sexuality;

  constructor(){}
}

File: ArtSiteApi/ArtSite.Data/Models/SearchObject.cs
using ArtSite.Data.Enums;

namespace ArtSite.Data.Models; 

public class SearchObject {
    public String? NameString { get; set; }
    public int? Species { get; set; }
    public Dictionary<int, bool>? Tags { get; set; }
    public int? MinAge { get; set; }
    public int? MaxAge { get; set; }
    public Gender? Gender { get; set; }
    public Sexuality? Sexuality { get; set; }
}